test/e2e/petset.go

```diff
@@ -243,6 +247,121 @@ var _ = framework.KubeDescribe("PetSet [Slow] [Feature:PetSet]", func() {
 	})
 })
 
+var _ = framework.KubeDescribe("Pet set recreate [Slow] [Feature:PetSet]", func() {
+	f := framework.NewDefaultFramework("pet-set-recreate")
+	var c *client.Client
+	var ns string
+
+	labels := map[string]string{
+		"foo": "bar",
+		"baz": "blah",
+	}
+	headlessSvcName := "test"
+	podName := "test-pod"
+	petSetName := "web"
+	petPodName := "web-0"
+
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("gce", "vagrant")
```

Comment:
Other tests are using persistent volumes and there was no dynamic storage provisioner for vagrant cluster. But i enabled hostpath provisioner (https://github.com/kubernetes/kubernetes/pull/32264) and now all petset tests should pass with 1 node vagrant cluster for sure.

I will try them and submit PR.